lighthouse-cli/run.js

```diff
@@ -218,10 +218,23 @@ async function runLighthouse(url, flags, config) {
     await potentiallyKillChrome(launchedChrome);
     process.removeListener('unhandledRejection', handleTheUnhandled);
 
+    // Runtime errors indicate something was *very* wrong with the page result.
+    // We don't want the user to have to parse the report to figure it out, so we'll still exit
+    // with an error code after we saved the results.
+    if (runnerResult && runnerResult.lhr.runtimeError) {
+      const {runtimeError} = runnerResult.lhr;
+      return printErrorAndExit({
+        name: 'LHError',
+        friendlyMessage: runtimeError.message,
+        lhrRuntimeError: true,
+        ...runtimeError,
```

Comment:
what's the use of these two lines since the lhr doesn't give this type anything else and we're just printing the friendlyMessage and exiting?

Reply:
typechecking and a theoretical future of using `.code` to be more discerning, it's just code and message though so I'll make it explicit